packages/jest-environment-jsdom/src/index.js

```diff
@@ -18,15 +18,20 @@ const ModuleMocker = require('jest-mock');
 
 class JSDOMEnvironment {
 
+  console: Object;
   document: ?Object;
   fakeTimers: ?FakeTimers;
   global: ?Global;
   moduleMocker: ?ModuleMocker;
 
-  constructor(config: Config): void {
+  constructor(config: Config, createJestConsole: (x: number) => Object): void {
```

Comment:
I think we could avoid breaking change by making `createJestConsole` optional (I'd rename it to `customConsole`). You could extract `createJestConsole` to a separate module and use it as a default.